src/test/java/pages/OrderConfirmationPage.java

```diff
@@ -0,0 +1,213 @@
+        datePickerField.click();
+    }
+
+    public void timePickerAccess()  {
+        timePickerField.click();
+    }
+
+    public void calendarAcceptChanges() {
+        calendarClockOkBtn.click();
+    }
+
+    public void setYearCalendar()  {
+        String date;
+        int year;
+        int desiredDateRealYear = getDesiredDateRealYear(desiredDate.getYear());
+
+        if(!presentYear)    {
+            if(!desiredDateIsFuture)  {
+                do
+                {
+                    calendarLeftBtn.click();
+                    date = dateCurrentYearMonth.getAttribute("innerText");
+                    year = Integer.parseInt(date.replaceAll("\\D", ""));
+                }   while (year != desiredDateRealYear);
+            }
+
+            if (desiredDateIsFuture) {
+                do
+                {
+                    calendarRightBtn.click();
+                    date = dateCurrentYearMonth.getAttribute("innerText");
+                    year = Integer.parseInt(date.replaceAll("\\D", ""));
+                }   while (year != desiredDateRealYear);
+            }
+        }
+    }
+
+    public void setMonthCalendar()  {
+        String month;
+        String desiredDateMonth = DateHandler.getMonthFromDate(desiredDate.getMonth());
+        String date;
+
+        if (!presentMonth)  {
+            if (!desiredDateIsFuture)   {
+                do
+                {
+                    calendarLeftBtn.click();
+                    date = dateCurrentYearMonth.getAttribute("innerText");
+                    month = date.replaceAll("[^A-z]", "");
+                }   while (!month.equals(desiredDateMonth));
+            }
+
+            if (desiredDateIsFuture)   {
+                do
+                {
+                    calendarRightBtn.click();
+                    date = dateCurrentYearMonth.getAttribute("innerText");
+                    month = date.replaceAll("[^A-z]", "");
+                }   while (!month.equals(desiredDateMonth));
+            }
+        }
+    }
+
+    public void setDayCalendar()    {
+        driver.findElement(By.xpath("//*[text() = '" + (desiredDate.getDay() + 1) + "']")).click();
+    }
+
+    public void setAM() {
+        amBtn.click();
+    }
+
+    public void setPM() {
+        pmBtn.click();
+    }
+
+    public void setHours()  {
+        action.dragAndDrop(clockTimePicker,
+                driver.findElement(By.xpath("//div[contains(@class, 'PickersClock')]" +
+                        "//span[text()=" + DateHandler.getHours(desiredDate) + "]")))
+                .build()
+                .perform();
+    }
+
+    public void setMinutes()    {
+        action.moveToElement(clockTimePicker,
+                        (int)Math.sin(DateHandler.getMinutes(desiredDate) * 6) * 100,
+                (int)Math.cos(DateHandler.getMinutes(desiredDate) * 6) * 100)
+                .click()
+                .perform();
+    }
+
+    public String getVolumeByItemName(String itemName) {
```

Comment:
If I understood the method getVolumeByItemName(String itemName) correctly, it might be better to use it in tests. We can compare the desired volume with the current one and write Assert in test.

Reply:
With this method we can pull the info that we need from our pages